cpp/ql/src/semmle/code/cpp/ir/dataflow/DefaultTaintTracking.qll

```diff
@@ -257,18 +257,6 @@ private predicate instructionTaintStep(Instruction i1, Instruction i2) {
   i2.(ChiInstruction).getPartial() = i1.(WriteSideEffectInstruction) and
   not i2.isResultConflated()
   or
-  // Flow from an element to an array or union that contains it.
-  i2.(ChiInstruction).getPartial() = i1 and
-  not i2.isResultConflated() and
-  exists(Type t | i2.getResultLanguageType().hasType(t, false) |
-    t instanceof Union
-    or
-    t instanceof ArrayType
-    or
-    // Buffers of unknown size
-    t instanceof UnknownType
-  )
-  or
```

Comment:
I think you can remove the same rule from IR `TaintTrackingUtil.qll`.

Reply:
Good point. I've removed it in 0b97a4a182bc19a9d32cea6364cc3e051be439a0.